tasks/managed/rh-sign-image/rh-sign-image.yaml

```diff
@@ -266,40 +251,81 @@ spec:
                     sourceTag=${tag}-source
 
                     if ! grep -q "^${registry_reference}:${sourceTag}$" "/tmp/${sourceContainerDigest}" ; then
-                      while (( ${RUNNING_JOBS@P} >= "$CONCURRENT_LIMIT" )); do
-                        wait -n
-                      done
-
-                      echo "Creating ${requestType} to sign image with tag ${tag}:"
-                      echo "- reference=${registry_reference}:${sourceTag}"
-                      echo "- manifest_digest=${sourceContainerDigest}"
-                      echo "- requester=$(params.requester)"
-
-                      ${requestType} -r "${request}" \
-                        -p pipeline_image="${pipeline_image}" \
-                        -p reference="${registry_reference}:${sourceTag}" \
-                        -p manifest_digest="${sourceContainerDigest}" \
-                        -p requester="$(params.requester)" \
-                        -p config_map_name="${config_map_name}" \
-                        -p taskGitUrl="$(params.taskGitUrl)" \
-                        -p taskGitRevision="$(params.taskGitRevision)" \
-                        -l ${TASK_LABEL}="${TASK_ID}" \
-                        -l ${PIPELINERUN_LABEL}="$(params.pipelineRunUid)" \
-                        -t "$(params.requestTimeout)" --pipeline-timeout "0h30m0s" --task-timeout "0h25m0s" \
-                        "${EXTRA_ARGS[@]}" -s true &
+                      to_sign_references+=("${registry_reference}:${sourceTag}")
+                      to_sign_digests+=("${manifest_digest}")
                     else
                       echo "Signature already exists for:"
                       echo "- reference=${registry_reference}:${sourceTag}"
                       echo "- manifest_digest=${sourceContainerDigest}"
                     fi
-
-                  ((++REQUEST_COUNT))
-                  echo "Request Count: $REQUEST_COUNT"
                 done
               done
             fi
         done
 
+        references_batch=""
+        digests_batch=""
+
+        # Process to sign arrays in batches
+        for i in "${!to_sign_references[@]}"; do
+          new_references_batch="${references_batch}${to_sign_references[$i]} "
+          new_digests_batch="${digests_batch}${to_sign_digests[$i]} "
+
+          # if batches are too big, send the request
```

Comment:
I wonder if this is actually correct. Maybe it is, just checking. What I found is only a limit for a taskrun result and that is 4096 bytes total. But it seems there is no specific limit for param size? I am not sure if the result size is an issue here. Maybe you were able to do some testing to validate this? I guess even if this was technically not needed, it's probably a good idea to limit the batches to some size?

Reply:
So it looks like, parameters are internally passed to pods as env variables and therefore limit is 32KB - according to gpt. But I haven't verified this

Reply:
OK. I guess we can stick with the 4096 size for now and tweak in the future if needed.